src/IlkRegistry.sol

```diff
@@ -85,10 +86,10 @@ contract IlkRegistry {
         _;
     }
 
-    VatLike  public vat;
-    CatLike  public cat;
-    SpotLike public spot;
-    GemInfo  private gemInfo;
+    VatLike  public immutable vat;
+    CatLike  public immutable cat;
```

Comment:
Since we're going to upgrade `cat` soon, what would be the implications of allowing this to be upgraded with a `file()`?  AFAICT, we cache everything in the registry, it would be a shame to have to redeploy because of a `cat` upgrade.

@gbalabasquer